design/secretless-e2e-tests-dnd.md

```diff
@@ -0,0 +1,276 @@
+3. [`juxtaposer`](https://github.com/cyberark/secretless-broker/tree/master/bin/juxtaposer), the Secretless perfomance testing framework, will eventually need to be automated. This is not addressed as part of this effort
+
+# Experience
+
+## Assumptions
+
+1. Moving the lifecycle management of dependency services behind an opaque API is a task that can be achieved with minimal effort. This is supported by the fact that the test run currently works while poluting the test source with service provisioning logic. The act of moving this logic behind an opaque API could be reduced to housing it in a seperate repository, that would still be an improvement. In this effort we go several steps further and take the opportunity to create a robust abstract with additional benefits. If at some point that becomes unwiedly we are always at liberty to descope the work and be less ambitious with the API.
+2. Adopting a test runner for writing and runing automated tests, given configured dependency resources (such as Conjur and target services), can be achieved with minimal effort. This is supported by the fact that there is currently only one path/test excercised as part of `kubernetes-conjur-demo` and that we only intend to add one more for now.
+
+## Overview
+
+We begin with an idea for a test.
+1. A developer writes the test case in as close to declarative a fashion as possible, leveraging preexisting steps and assertions; defining new steps and assertions if necessary.
+2. A developer specifies the service dependencies for their test case using the predefined test cases that come with the Service Engine (defined in the Architecture section).
+3. A developer runs a simple command to run the tests, this results in either success or failure, in both cases the outcome is reported and any resources used in the test are cleaned up (unless requested to stay). In the case of failure breadcrumbs are provided fo where in the test the failure occurred.
+4. A developer adds the test to CI via Jenkinsfile and pushes the branch to Github
+5. The tests run automatically as in Step #3, this time on CI infrastructure. 
+
+## Walkthroughs
+
+TBD
+
+# Technical Design
+
+## Architecture
+
+A `Service` is any external software that end to end tests rely on e.g. Conjur, Postgres etc.
+
+A `Service Config` is a JSON object that specifies information required for a service to be created e.g. for Conjur `CONJUR_ACCOUNT`, `version` etc. Conjur will have different configuration options to Postgres. These options must be documented for each Service. Ideally this would be in a centralized repository. It's important for all users of a Service to know what they're able to configure on a service.
+```
```

Comment:
I'd probably guess that yaml might be a bit more in line with our other stuff 🤷‍♂